lib/widgets-api.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Functions for working with widgets in WordPress. These should ultimately live
+ * in wp-includes/widgets.php, wp-admin/includes/widgets.php, etc. when merged
+ * to Core.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Assigns a widget to the given sidebar.
+ *
+ * Belongs in wp-includes/widgets.php when merged to Core.
+ *
+ * @since 10.2.0
+ *
+ * @param string $widget_id  The widget id to assign.
+ * @param string $sidebar_id The sidebar id to assign to. If empty, the widget won't be added to any sidebar.
+ */
+function gutenberg_assign_widget_to_sidebar( $widget_id, $sidebar_id ) {
```

Comment:
How would this go into core?

Reply:
It would live in `wp-includes/widgets.php` and be named `wp_assign_widget_to_sidebar()`.